changelog/@unreleased/pr-848.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: fix
+fix:
+  description: Replace liche with lychee for markdown link checking
+  links:
+  - https://github.com/palantir/conjure/pull/848
```

Comment:
I'd have probably used the `no changelog` tag for this PR since it doesn't impact consumers.